### DISCUSSION TITLE
Set `use-uv` per project

### DISCUSSION THREAD

**Discussion by olliethomas** [astral-sh/rye#742] [General]
Hi,

I've just started using rye - it's very nice, thank you. 
There are some projects I have where experimenting with `uv` is suitable, and others where I would really rather use `pip-tools`. Is there a way of setting the `use-uv` behaviour in a pyproject.toml rather than than in the global rye config?

I guess this could be a broader question of how to override any behaviour per project. Is that something that's possible?

Thanks for your help

**Comment by mitsuhiko** [comment]
A lot of behavior can be overridden per project, this is not yet possible for `uv`. Maybe `uv` catches up fast enough that we can phase out `pip-tools` but I guess that might need a little bit longer.

**Comment by eeriemyxi** [comment]
> A lot of behavior can be overridden per project, this is not yet possible for `uv`. Maybe `uv` catches up fast enough that we can phase out `pip-tools` but I guess that might need a little bit longer.

This is sad. `pip-tools` is very slow on my machine, but one of my active projects does not build when `uv` is on. The package that fails to build is `"sixel @ git+https://github.com/lubosz/python-sixel.git"`. `pip` can compile the package with `GCC` but `uv` can't. However, `uv` did want to build it using `clang` which I don't have installed because I have no use for it when I have GCC already. So, to remedy that, I ran exported `CC=gcc`; which did make it work, however, the build failed with a lot of errors from the compiler.

The issue is that all my other projects work just fine with `uv` and it's much much much faster than `pip-tools`.
```
myxi :: coding/python/lyto ‹main› » time rye sync
Reusing already existing virtualenv
Generating production lockfile: /home/myxi/Documents/coding/python/lyto/requirements.lock
Generating dev lockfile: /home/myxi/Documents/coding/python/lyto/requirements-dev.lock
Installing dependencies
Found existing installation: lyto 0.2.0
...
Successfully built lyto sixel
Installing collected packages: sixel, lyto
Successfully installed lyto-0.2.2 sixel-0.2.0
Done!
rye sync  53.38s user 6.32s system 80% cpu 1:14.52 total
```
*(average time of `rye sync` with `pip-tools` on my machine)*
